presidio-analyzer/presidio_analyzer/predefined_recognizers/azure_ai_language.py

```diff
@@ -0,0 +1,155 @@
+import os
+from typing import List, Optional
+import logging
+
+try:
+    from azure.ai.textanalytics import TextAnalyticsClient
+    from azure.core.credentials import AzureKeyCredential
+
+except ImportError:
+    TextAnalyticsClient = None
+    AzureKeyCredential = None
+from presidio_analyzer import EntityRecognizer, RecognizerResult, AnalysisExplanation
+from presidio_analyzer.nlp_engine import NlpArtifacts
+
+logger = logging.getLogger("presidio-analyzer")
+
+
+class AzureAILanguageRecognizer(EntityRecognizer):
+    """Wrapper for PII detection using Azure AI Language."""
+
+    def __init__(
+        self,
+        supported_entities: Optional[List[str]] = None,
+        supported_language: str = "en",
+        ta_client: Optional[TextAnalyticsClient] = None,
+        azure_ai_key: Optional[str] = None,
+        azure_ai_endpoint: Optional[str] = None,
+    ):
+        """
+        Wrapper for the PII detection in Azure AI Language
+        :param supported_entities: List of supported entities for this recognizer.
+        If None, all supported entities will be used.
+        :param supported_language: Language code to use for the recognizer.
+        :param ta_client: object of type TextAnalyticsClient. If missing,
+        the client will be created using the key and endpoint.
+        :param azure_ai_key: Azure AI  for language key
+        :param azure_ai_endpoint: Azure AI for language endpoint
+
+        For more info, see https://learn.microsoft.com/en-us/azure/ai-services/language-service/personally-identifiable-information/overview # noqa
+        """
+
+        super().__init__(
+            supported_entities=supported_entities,
+            supported_language=supported_language,
+            name="Azure AI Language PII",
+        )
+
+        is_available = bool(TextAnalyticsClient)
+        if not is_available:
+            raise ValueError(
+                "Azure AI Language is not available. "
+                "Please install the required dependencies:"
+                "1. azure-ai-textanalytics"
+                "3. azure-core"
+            )
+
+        if not supported_entities:
+            self.supported_entities = self.get_azure_ai_supported_entities()
+
+        azure_ai_key = (
+            azure_ai_key if azure_ai_key else os.getenv("AZURE_AI_KEY", None)
+        )
+        azure_ai_endpoint = (
+            azure_ai_endpoint
+            if azure_ai_endpoint
+            else os.getenv("AZURE_AI_ENDPOINT", None)
+        )
+        if azure_ai_key is None:
```

Comment:
nit: you can do os.environ['AZURE_AI_KEY'] instead of getenv and catch a KeyError

Reply:
The current implementation catches Nones from either the user or the env. Wouldn't we only catch a missing env variable if we catch a KeyError?

Reply:
If its none from the user, it would go to the environ and raise a key error, wont it give the same if catch the key error and raise the one you are raising now?

Reply:
Not sure I follow :) Are you suggesting to try/catch a KeyError instead of doing if and raise ValueError?